agent/api/task/task.go

```diff
@@ -479,6 +489,53 @@ func (task *Task) configureContainerDependenciesForServiceConnect() {
 	}
 }
 
+func (task *Task) initServiceConnectEphemeralPorts() error {
+	var utilizedPorts []uint16
+	// First determine how many ephemeral ports we need
+	var numEphemeralPortsNeeded int
+	for _, ic := range task.ServiceConnectConfig.IngressConfig {
+		if ic.ListenerPort == 0 { // This means listener port was not sent to us by ACS, signaling the port needs to be ephemeral
+			numEphemeralPortsNeeded++
+		} else {
+			utilizedPorts = append(utilizedPorts, ic.ListenerPort)
+		}
+	}
+
+	// Presently, SC egress port is always ephemeral, but adding this for future-proofing
+	if task.ServiceConnectConfig.EgressConfig.ListenerPort == 0 {
```

Comment:
Do we want `ListenerPort` to be defined in the `EgressConfig` model as well?

Reply:
Yes, this is the internal model, not the ACS model. I want to keep all the SC info in one place.

Reply:
I see. That makes sense.